ehsm_kms_service/delete_secret_thread.js

```diff
@@ -0,0 +1,138 @@
+const {
+    base64_decode
+} = require('./function')
+
+// Example of setInterval
+var init_interval
+
+// Record timer waiting to run
+var timer_queue = {}
+
+/**
+ * remove a timer from exist timer by key
+ * @param {Object} DB : database control
+ * @param {Object} key : key form exist timer
+ * @returns
+ */
+const removeTimer = (secretName, appid, DB) => {
+    try {
+        //Delete thread when thread exists
+        if (timer_queue[secretName + '_' + appid] != undefined ) {
+            clearTimeout(timer_queue[secretName + '_' + appid].timer)
+        }
+        delete timer_queue[secretName + '_' + appid];
+        // An insurance mechanism to ensure that thread init_interval is active
+        init_secret_delete_thread(DB)
+    } catch (error) {
+        console.error('error removeTimer :: ', error)
+    }
+}
+
+/**
+ * Expose the interface externally, scheduled tasks of delete secret
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database control
+ * @param {Object} secretName : The name of the secret. eg. 'secretName01'
+ * @param {Object} plannedDeleteTime : planned delete time for secret eg. '1659931644416'
+ * @returns
+ */
+const startDeleteSecretTask = (DB, appid, secretName, plannedDeleteTime) => {
+    try {
```

Comment:
based on my understanding, seems this function is used to add a new delete task into the delete queue with the secret name.
if yes, suggest to rename as add_delete_task(),  and the corresponding function of removeTimer renamed as remove_delete_task().

and the timer_queue renamed as delete_task_queue